### PR TITLE
Make RabbitMQ retry connecting

### PR DESCRIPTION
**Pull request by jjkester**

### CODE REVIEW

File: implementation/pubsub/rabbitmq/common/src/main/java/org/inaetics/dronessimulator/pubsub/rabbitmq/common/RabbitConnection.java [hunk 1]
```diff
@@ -93,12 +99,25 @@ public void setConfig(Dictionary<String, String> config) {
...
+                    getLogger().info("Connected to RabbitMQ");
+                } catch (ConnectException | TimeoutException e) {
+                    getLogger().error("Could not connect to RabbitMQ (attempt {}): {}", attempt, e.getMessage());
+
+                    if (attempt >= MAX_CONNECTION_ATTEMPTS) {
+                        throw new IOException(e);
+                    } else {
+                        try {
+                            Thread.sleep(CONNECTION_ATTEMPT_TIMEOUT);
+                        } catch (InterruptedException ignored) {}
```
Comment (ghost) [inline implementation/pubsub/rabbitmq/common/src/main/java/org/inaetics/dronessimulator/pubsub/rabbitmq/common/RabbitConnection.java]: Throw new exception as service might have been stopped. Or just do not catch. Right now, service cannot be stopped as thread may not be interrupted

File: implementation/pubsub/rabbitmq/common/src/main/java/org/inaetics/dronessimulator/pubsub/rabbitmq/common/RabbitConnection.java [hunk 2]
```diff
@@ -22,6 +22,12 @@
  * related to the connection to the broker.
  */
 public abstract class RabbitConnection {
+    /** The maximum number of connection attempts. */
+    private static final int MAX_CONNECTION_ATTEMPTS = 10;
+
+    /** The timeout between connection attempts in milliseconds. */
+    private static final long CONNECTION_ATTEMPT_TIMEOUT = 10000;
```
Comment (ghost) [inline implementation/pubsub/rabbitmq/common/src/main/java/org/inaetics/dronessimulator/pubsub/rabbitmq/common/RabbitConnection.java]: Lower sleep time to 2 seconds en increase attempts to 40? This allows for a more responsive experience.

**Review by ghost — CHANGES_REQUESTED** [review]

File: implementation/pubsub/rabbitmq/common/src/main/java/org/inaetics/dronessimulator/pubsub/rabbitmq/common/RabbitConnection.java (hunk 1 above)
Comment (jjkester) [reply implementation/pubsub/rabbitmq/common/src/main/java/org/inaetics/dronessimulator/pubsub/rabbitmq/common/RabbitConnection.java]: Did not think about this piece of code, IntelliJ suggested to catch the exception so I did. Will fix this.

File: implementation/pubsub/rabbitmq/common/src/main/java/org/inaetics/dronessimulator/pubsub/rabbitmq/common/RabbitConnection.java (hunk 2 above)
Comment (jjkester) [reply implementation/pubsub/rabbitmq/common/src/main/java/org/inaetics/dronessimulator/pubsub/rabbitmq/common/RabbitConnection.java]: Yeah, sure.

File: implementation/pubsub/rabbitmq/common/src/main/java/org/inaetics/dronessimulator/pubsub/rabbitmq/common/RabbitConnection.java (hunk 1 above)
Comment (jjkester) [reply implementation/pubsub/rabbitmq/common/src/main/java/org/inaetics/dronessimulator/pubsub/rabbitmq/common/RabbitConnection.java]: Decided to catch the exception and let the method return, so it just exists gracefully.

**Pull request merged**